sim-lib/src/lib.rs

```diff
@@ -1034,6 +1134,7 @@ async fn produce_simulation_results(
     while let Some(res) = set.join_next().await {
         if let Err(e) = res {
             log::error!("Simulation results producer task exited with error: {e}.");
+            shutdown.trigger();
```

Comment:
Do we need to trigger shutdown here? Looks like the loop above only exits on `listener.clone`, so we've already had a shutdown trigger elsewhere?

Reply:
In hindsight, it was not necessary. I have update the handling of the error here (and across all tasks) to follow the recommendation [here](https://github.com/bitcoin-dev-project/sim-ln/pull/164#issuecomment-1919135888).